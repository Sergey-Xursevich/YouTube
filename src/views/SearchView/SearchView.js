export default class SearchView {
  render() {
    const { body } = document;
    const search = document.createElement('div');
    const inputSearch = document.createElement('input');
    const btnSearch = document.createElement('button');

    body.appendChild(search);
    search.classList.add('search');

    inputSearch.setAttribute('type', 'text');
    inputSearch.setAttribute('placeholder', 'What kind of video do you want to find?');

    btnSearch.setAttribute('type', 'submit');
    btnSearch.classList.add('btn');

    search.appendChild(inputSearch);
    search.appendChild(btnSearch);
  }
}
