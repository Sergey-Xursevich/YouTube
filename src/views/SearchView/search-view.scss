* {
  box-sizing: border-box;
}

body {
  margin: 0;
  background: #343D46;
}

.search {
  margin-top: 100px;
  position: relative;
  width: 90%;
  margin: 80px auto;

  input {
      width: 100%;
      height: 42px;
      padding-left: 10px;
      border: 2px solid white;
      border-radius: 5px;
      color: rgb(104, 93, 93);
      font-size: 18px;
      outline: none;
      &:focus {
        border: 3px solid red;
      }
    }
  }

.btn {
  position: absolute;
  top: 0;
  right: 0px;
  width: 60px;
  height: 100%;
  border: none;
  background: coral;
  border-radius: 0 5px 5px 0;
  cursor: pointer;

  &:before {
      content: "\f002";
      font-family: 'FontAwesome';
      font-size: 16px;
      color: #F9F0DA;
  }

  &:focus {
    outline: none;
  }
}
