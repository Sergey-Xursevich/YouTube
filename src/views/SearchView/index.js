import SearchView from './SearchView';

import './search-view.scss';

export default SearchView;
