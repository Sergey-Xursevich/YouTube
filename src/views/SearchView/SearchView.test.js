import SearchView from './SearchView';

describe('Slider render', () => {
  it('Should be an instance of Function', () => {
    expect(SearchView.prototype.render).toBeInstanceOf(Function);
  });
});


describe('Should be an render correctly', () => {
  SearchView.prototype.render();
  expect(document.body.innerHTML).toMatchSnapshot();
});
