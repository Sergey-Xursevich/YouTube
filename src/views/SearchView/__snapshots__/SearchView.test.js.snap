// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"<div class=\\"search\\"><input type=\\"text\\" placeholder=\\"What kind of video do you want to find?\\"><button type=\\"submit\\" class=\\"btn\\"></button></div>"`;
